sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/message_sender.hpp

```diff
@@ -176,7 +181,12 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     _azure_NODISCARD std::tuple<MessageSendStatus, Models::_internal::AmqpError> Send(
         Models::AmqpMessage const& message,
         Context const& context = {});
+#elif ENABLE_RUST_AMQP
+    _azure_NODISCARD Models::_internal::AmqpError Send(
+        Models::AmqpMessage const& message,
+        Context const& context = {});
 
+#endif
```

Comment:
It would be helpful to understand with a comment in source why this method ends up only available in the rust case, while the uamqp version is slightly different. Basically, what's causing some of these #ifs/elifs between implementations.
Why does one return AmqpError only while the other return the MessageSendStatus too?